src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java

```diff
@@ -0,0 +1,111 @@
+          if (save.failed()) {
+            logger.error(save.cause());
+            if (ValidationHelper.isDuplicate(save.cause().getMessage())) {
+              asyncResultHandler.handle(Future.succeededFuture(
+                PostPatronNoticePolicyStoragePatronNoticePoliciesResponse
+                  .respond422WithApplicationJson(createNotUniqueNameErrors(entity.getName()))));
+            }
+            asyncResultHandler.handle(Future.succeededFuture(
+              PostPatronNoticePolicyStoragePatronNoticePoliciesResponse.respond500WithTextPlain(save.cause())));
+            return;
+          }
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostPatronNoticePolicyStoragePatronNoticePoliciesResponse.respond201WithApplicationJson(entity)));
+        });
+      } catch (Exception e) {
+        logger.error(e);
+        asyncResultHandler.handle(Future.succeededFuture(
+          PostPatronNoticePolicyStoragePatronNoticePoliciesResponse.respond500WithTextPlain(e)));
+      }
+    });
+  }
+
+  @Override
+  public void putPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyId(
+    String patronNoticePolicyId,
+    PatronNoticePolicy entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    vertxContext.runOnContext(v -> {
+      try {
+        String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+        PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+
+        pgClient.update(PATRON_NOTICE_POLICY_TABLE, entity, patronNoticePolicyId, update -> {
+          if (update.failed()) {
+            logger.error(update.cause());
+            if (ValidationHelper.isDuplicate(update.cause().getMessage())) {
+              asyncResultHandler.handle(Future.succeededFuture(
+                PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
+                  .respond422WithApplicationJson(createNotUniqueNameErrors(entity.getName()))));
+            }
+            asyncResultHandler.handle(Future.succeededFuture(
+              PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
+                .respond500WithTextPlain(update.cause())));
+            return;
+          }
+          asyncResultHandler.handle(Future.succeededFuture(
+            PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.respond204()));
+        });
+      } catch (Exception e) {
+        logger.error(e);
+        asyncResultHandler.handle(Future.succeededFuture(
+          PutPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.respond500WithTextPlain(e)));
+      }
+    });
+  }
+
+  private Errors createNotUniqueNameErrors(String name) {
+    Error error = new Error();
+    error.setMessage(String.format("'%s' name in not unique", name));
+    error.setCode(STATUS_CODE_DUPLICATE_NAME);
```

Comment:
Please see [comment on the issue](https://issues.folio.org/browse/CIRCSTORE-89?focusedCommentId=42707&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-42707) regarding the use of code for error identification